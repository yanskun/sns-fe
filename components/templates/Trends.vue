<template lang="pug">
  v-main.trend-root
    post-trend-modal(
      :options="options"
      :ogp="postOgp"
      @click-cancel="$emit('click-modal-cancel')"
      @click-post="$emit('click-modal-post')"
      @input-url="$emit('input-modal-url', $event)"
    )
    .trend-list
      trend-card.card(
        v-for="trendCard in trends"
        :key="trendCard.id"
        :tags="trendCard.tags"
        :comment="trendCard.comment"
        :ogp="trendCard.ogp"
      )
</template>

<script lang="ts">
import { defineComponent, PropType } from '@vue/composition-api'
import TrendCard, { Tag, Ogp } from '../organisms/TrendCard.vue'
import PostTrendModal, { Option } from '../organisms/PostTrendModal.vue'

export type Trend = {
  id: string
  tags: Tag[]
  ogp: Ogp
  comment: string
}

export default defineComponent({
  components: {
    TrendCard,
    PostTrendModal,
  },

  props: {
    trends: {
      type: Array as PropType<Trend[]>,
      required: false,
      default: () => [],
    },
    options: {
      type: Array as PropType<Option[]>,
      required: true,
    },
    postOgp: {
      type: Object as PropType<Ogp>,
      required: false,
      default: () => {
        return {
          href: '',
          src: '',
          title: '',
          description: '',
        }
      },
    },
  },
})
</script>

<style lang="scss" scoped>
.trend-root {
  height: 100%;
  width: 100%;
  overflow-y: scroll;
  .trend-list {
    max-width: 60%;
    margin: 0 auto;
    .card {
      margin: 20px;
    }
  }
}
</style>
