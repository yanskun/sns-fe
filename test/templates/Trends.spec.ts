import { createLocalVue, shallowMount } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import Vuetify from 'vuetify'
import Component from '~/components/templates/Trends.vue'
import { genTrends } from '~/stories/templates/Trends.stories'

const localVue = createLocalVue()
localVue.use(VueCompositionApi)
localVue.use(Vuetify)

describe('Trends', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(Component, {
      propsData: {
        trends: genTrends(3),
        options: [
          { text: 'front', value: 'front' },
          { text: 'server', value: 'server' },
        ],
        ogp: {
          href: 'https://github.com/',
          src: 'https://github.githubassets.com/images/modules/open_graph/github-logo.png',
          title: 'GitHub',
          description:
            'GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.',
        },
      },
      localVue,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
