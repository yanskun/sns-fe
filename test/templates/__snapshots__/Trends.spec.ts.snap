// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trends snapshot 1`] = `
<v-main-stub
  class="trend-root"
  ogp="[object Object]"
  tag="main"
>
  <post-trend-modal-stub
    ogp="[object Object]"
    options="[object Object],[object Object]"
  />
  <div
    class="trend-list"
  >
    <trend-card-stub
      class="card"
      comment="Nuxt.js is amazing"
      ogp="[object Object]"
      tags="[object Object],[object Object]"
    />
    <trend-card-stub
      class="card"
      comment="Nuxt.js is amazing"
      ogp="[object Object]"
      tags="[object Object],[object Object]"
    />
    <trend-card-stub
      class="card"
      comment="Nuxt.js is amazing"
      ogp="[object Object]"
      tags="[object Object],[object Object]"
    />
  </div>
</v-main-stub>
`;
